roles/provision_vm/tasks/clone_an_existing_vm.yml

```diff
@@ -0,0 +1,82 @@
+---
+- name: Fail if the VM is not exists
+  block:
+  - name: Retrieve all the details about the new VM
+    vmware.vmware_rest.vcenter_vm_info:
+      vcenter_hostname: "{{ connection_args.vcenter_hostname }}"
+      vcenter_username: "{{ connection_args.vcenter_username }}"
+      vcenter_password: "{{ connection_args.vcenter_password }}"
+      vcenter_validate_certs: "{{ connection_args.vcenter_validate_certs }}"
+      names:
+      - "{{ vm_name }}"
+    register: vms
+
+  - name: Fail when the VM is already exists
+    ansible.builtin.fail:
+      msg: The VM "{{ template }}" from is not exist
+    when: vms.value | length <1
+
+- name: Create a new VM
+  block:
+  - name: Create or update the VM
+    community.vmware.vmware_guest:
+      hostname: "{{ connection_args.vcenter_hostname }}"
+      username: "{{ connection_args.vcenter_username }}"
+      password: "{{ connection_args.vcenter_password }}"
+      validate_certs: "{{ omit if connection_args.vcenter_validate_certs is not defined else connection_args.vcenter_validate_certs}}"
+
+      port: "{{ omit if port is not defined else port }}"
+      proxy_host: "{{ omit if proxy_host is not defined else proxy_host }}"
+      proxy_port: "{{ omit if proxy_port is not defined else proxy_port }}"
+
+      name: "{{ vm_name }}"
+      uuid: "{{ omit if uuid is not defined else uuid }}"
+
+      # compute resource:
+      cluster: "{{ omit if cluster not defined else cluster }}"
+      esxi_hostname: "{{ omit if esxi_hostname is not defined else esxi_hostname}}"
+      datacenter: "{{ omit if datacenter is not defined else datacenter}}"
+      folder: "{{ omit if folder is not defined else folder}}"
+      datastore: "{{ omit if datastore is not defined else datastore}}"
+      resource_pool: "{{ omit if resource_pool is not defined else lookup('vmware.vmware_rest.resource_pool_moid', {{ datacenter_name }}/host/{{ cluster_name }}/Resources/{{ resource_pool_name }}, **connection_args) }}"
```

Comment:
I think this parameter don't expect moid, so so there is no need to use `resource_pool_moid`

Reply:
You're right